src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py

```diff
@@ -333,17 +338,20 @@ def refreshCache(self):
                     badJobs[71102].append(newJob)
                     continue
 
-            # try to remove draining sites if possible, this is needed to stop
-            # jobs that could run anywhere blocking draining sites
-            # if the job type is Merge, LogCollect or Cleanup this is skipped
-            if newJob['task_type'] not in self.ioboundTypes:
-                nonDrainingSites = [x for x in possibleLocations if x not in self.drainSites]
-                if nonDrainingSites:  # if >1 viable non-draining site remove draining ones
-                    possibleLocations = nonDrainingSites
-                else:
-                    newJob['possibleSites'] = possibleLocations
-                    badJobs[71104].append(newJob)
-                    continue
+                # try to remove draining sites if possible, this is needed to stop
+                # jobs that could run anywhere blocking draining sites
+                # if the job type is Merge, LogCollect or Cleanup this is skipped
+                if newJob['task_type'] not in self.ioboundTypes:
+                    nonDrainingSites = [x for x in possibleLocations if x not in self.drainSites]
+                    if nonDrainingSites:  # if >1 viable non-draining site remove draining ones
+                        possibleLocations = nonDrainingSites
+                    else:
+                        newJob['possibleSites'] = possibleLocations
+                        badJobs[71104].append(newJob)
+                        continue
+            else:
+                logging.info("Agent is in speed drain mode. Submitting jobs to all possible locations.")
```

Comment:
This logging can't be here since we're iterating over every potential job to be submitted.
I'd place it around L280 
```
if self.enableAllSites:
    logging....
```